### PR TITLE
[AIR/Docs] Add Predictor Docs

### PR DESCRIPTION
**Pull request by amogkam**
## Why are these changes needed?

Adds documentation for batch inference and using a Predictor.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-air/batch-prediction.rst
Comment (richardliaw) [inline doc/source/ray-air/batch-prediction.rst:1]: Merge this with predictors.rst

File: doc/source/ray-air/doc_code/predictors.py
Comment (ericl) [inline doc/source/ray-air/doc_code/predictors.py:73]: This is really bad. Can we push on merging https://github.com/ray-project/ray/pull/26603 ?

File: doc/source/ray-air/predictors.rst
Comment (ericl) [inline doc/source/ray-air/predictors.rst:6]: Nice diagram!

File: doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py
Comment (jiaodong) [inline doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py:23]: the `len() -1` is not needed if we dropped column here first ?

File: doc/source/ray-air/examples/tf_tabular_batch_prediction.py
Comment (jiaodong) [inline doc/source/ray-air/examples/tf_tabular_batch_prediction.py:24]: ditto

File: doc/source/ray-air/examples/torch_image_batch_pretrained.py
Comment (jiaodong) [inline doc/source/ray-air/examples/torch_image_batch_pretrained.py:17]: add todo to remove this once ImageFolder merged ? I had the same for my CUJ PRs and we just need to make sure all can be search and replaced in one shot https://github.com/ray-project/ray/pull/26614

File: doc/source/ray-air/doc_code/predictors.py
Comment (amogkam) [inline doc/source/ray-air/doc_code/predictors.py:32]: Does this work if you flatten the array?

File: doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py
Comment (amogkam) [inline doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py:23]: Can you use the feature_columns arg in batchpredictor instead of dropping the column?

File: doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py
Comment (amogkam) [inline doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py:36]: Should we also show how one should do scoring and get final accuracies from the predictions?

File: doc/source/ray-air/examples/tf_tabular_batch_prediction.py
Comment (amogkam) [reply doc/source/ray-air/examples/tf_tabular_batch_prediction.py:24]: Same here- can you use feature_columns instead of dropping?

Comment (richardliaw) [reply doc/source/ray-air/examples/tf_tabular_batch_prediction.py]: done

File: doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py
Comment (richardliaw) [reply doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py]: done

File: doc/source/ray-air/examples/torch_image_batch_pretrained.py
Comment (richardliaw) [reply doc/source/ray-air/examples/torch_image_batch_pretrained.py:18]: done

File: doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py
Comment (richardliaw) [reply doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py]: done

File: doc/source/ray-air/doc_code/predictors.py
Comment (richardliaw) [reply doc/source/ray-air/doc_code/predictors.py]: done

File: doc/source/ray-air/predictors.rst
Comment (richardliaw) [reply doc/source/ray-air/predictors.rst]: thanks! @Yard1 made it for a talk

File: doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py
Comment (richardliaw) [reply doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py:39]: I'd like to not block on this. Can you show me how to do this?

I tried to update the example to do this but:
```python

predicted_probabilities = batch_predictor.predict(
    dataset, feature_columns=features, keep_columns=["target"])
predicted_probabilities.show()
# {'predictions': array([1.], dtype=float32)}
# {'predictions': array([0.], dtype=float32)}

# Calculate final accuracy.
def calculate_accuracy(df):
   return pd.DataFrame({"correct": df["predictions"] == df["target"]})

correct = predicted_probabilities.map_batches(calculate_accuracy)
print("Final accuracy: ", correct.sum(on="correct") / correct.count())
```

But i'm getting:
```python
(map pid=91224)   File "pyarrow/_compute.pyx", line 327, in pyarrow._compute.Function.call
(map pid=91224)   File "pyarrow/error.pxi", line 143, in pyarrow.lib.pyarrow_internal_check_status
(map pid=91224)   File "pyarrow/error.pxi", line 120, in pyarrow.lib.check_status
(map pid=91224) pyarrow.lib.ArrowNotImplementedError: Function sum has no kernel matching input types (array[extension<arrow.py_extension_type<ArrowTensorType>>])
```

File: doc/source/ray-air/doc_code/predictors.py
Comment (richardliaw) [reply doc/source/ray-air/doc_code/predictors.py:73]: yeah, approved.

Comment (richardliaw) [reply doc/source/ray-air/doc_code/predictors.py:73]: with UTs addd

File: doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py
Comment (amogkam) [reply doc/source/ray-air/examples/pytorch_tabular_batch_prediction.py:39]: Made a new PR https://github.com/ray-project/ray/pull/26636/ which adds this section.

The changes are here https://github.com/ray-project/ray/pull/26636/commits/85b7b088fb7a487768dacf0f02107452d11de0d8.

We can either merge that into this one, or if you would like to not block on it, then we can merge that in as a follow up.

**Review by richardliaw — APPROVED** [review]

File: doc/source/ray-air/doc_code/predictors.py
Comment (richardliaw) [inline doc/source/ray-air/doc_code/predictors.py:61]:
```suggestion
    return pd.DataFrame({"correct": int(df["predictions"][0]) == df["label"]})

```

File: doc/source/ray-air/doc_code/predictors.py
Comment (richardliaw) [inline doc/source/ray-air/doc_code/predictors.py:64]:
```suggestion
    return pd.DataFrame({"correct": int(df["predictions"][0]) == df["label"]})


predictions = batch_predictor.predict(
    ds, feature_columns=["feature_1"], keep_columns=["label"]
```

**Pull request merged**
